t/lib/OpenQA/Test/FullstackUtils.pm

```diff
@@ -78,14 +78,18 @@ sub reload_manually {
     prevent_reload($driver);
 }
 
+sub find_status_text { shift->find_element('#result-row .card-body')->get_text() }
+
 sub wait_for_result_panel {
     my ($driver, $result_panel, $desc, $fail_on_incomplete, $refresh_interval) = @_;
     $refresh_interval //= 1;
 
     prevent_reload($driver);
 
+    # Report failure at the callsite instead of the test function
+    local $Test::Builder::Level = $Test::Builder::Level + 1;
```

Comment:
I guess this does not help in all cases, e.g. when this is called from `wait_for_job_running` which is also within this helper module.